hazelcast/include/hazelcast/client/spi/impl/AbstractInvocationFuture.h

```diff
@@ -0,0 +1,534 @@
+                            }
+                        }
+                    }
+
+                    virtual boost::shared_ptr<T> get(int64_t timeout, const TimeUnit &unit) {
+                        boost::shared_ptr<ThreadState> thread = boost::shared_ptr<ThreadState>(
+                                new ThreadState(util::getCurrentThreadId()));
+                        boost::shared_ptr<BaseState> response = registerWaiter(thread, boost::shared_ptr<Executor>());
+                        if (response != VOIDOBJECT) {
+                            return resolveAndThrowIfException(response);
+                        }
+
+                        int64_t deadlineNanos = util::currentTimeNanos() + unit.toNanos(timeout);
+                        int64_t timeoutNanos = unit.toNanos(timeout);
+                        while (timeoutNanos > 0) {
+                            thread->parkNanos(timeoutNanos);
+                            timeoutNanos = deadlineNanos - util::currentTimeNanos();
+
+                            if (isDone()) {
+                                return resolveAndThrowIfException(state);
+                            }
+                        }
+
+                        unregisterWaiter(thread);
+                        throw (ExceptionBuilder<TimeoutException>("AbstractInvocationFuture::get(timeout, unit)")
+                                << "Timeout: " << unit.toMillis(timeout) << " msecs").build();
+                    }
+
+                    /**
+                     * Can be called multiple times, but only the first answer will lead to the
+                     * future getting triggered. All subsequent complete calls are ignored.
+                     *
+                     * @param value The type of response to offer.
+                     * @return <tt>true</tt> if offered response, either a final response or an
+                     * internal response, is set/applied, <tt>false</tt> otherwise. If <tt>false</tt>
+                     * is returned, that means offered response is ignored because a final response
+                     * is already set to this future.
+                     */
+                    virtual bool complete(const boost::shared_ptr<T> &value) {
+                        boost::shared_ptr<BaseState> newState = boost::shared_ptr<BaseState>(new ValueState(value));
+
+                        return innerComplete(newState);
+                    }
+
+                    virtual bool complete(const boost::shared_ptr<exception::IException> &exception) {
+                        boost::shared_ptr<BaseState> newState = boost::shared_ptr<BaseState>(
+                                new ExceptionState(exception));
+
+                        return innerComplete(newState);
+                    }
+
+                protected:
+                    typedef ExecutionCallback<T> CALLBACKTYPE;
+                    
+                    class BaseState {
+                    public:
```

Comment:
Missing virtual destructor

Reply:
I has no virtual method and hence should not need the virtual destructor, does it?